src/components/swap-history/index.tsx

```diff
@@ -112,14 +112,45 @@ const Transaction: FC<{ address: string; transaction: TransactionProps }> = ({
 const Component: FC = () => {
   const { t } = useTranslation();
   const { address = "", isConnected } = useAccount();
-  const { transactions } = useSwapHistory();
+  const { transactions, clearHistory } = useSwapHistory();
 
   return isConnected ? (
     <div className="swap-history">
-      <span className="heading">{t(constantKeys.TRANSACTIONS)}</span>
-      {transactions.map((transaction, index) => (
-        <Transaction key={index} address={address} transaction={transaction} />
-      ))}
+      <div className="history-header">
+        <span className="heading">{t(constantKeys.TRANSACTIONS)}</span>
+        {transactions.length > 0 && (
+          <button 
+            className="clear-history-btn" 
+            onClick={clearHistory}
+            aria-label="Clear transaction history"
+          >
+            <svg 
+              width="24" 
+              height="24" 
+              viewBox="0 0 24 24" 
+              fill="none" 
+              xmlns="http://www.w3.org/2000/svg"
+            >
+              <path 
+                d="M3 6H21M19 6V20C19 21 18 22 17 22H7C6 22 5 21 5 20V6M8 6V4C8 3 9 2 10 2H14C15 2 16 3 16 4V6M10 11V17M14 11V17" 
+                stroke="currentColor" 
+                strokeWidth="2" 
+                strokeLinecap="round" 
+                strokeLinejoin="round"
+              />
+            </svg>
+          </button>
+        )}
+      </div>
+      {transactions.length > 0 ? (
+        transactions.map((transaction, index) => (
+          <Transaction key={index} address={address} transaction={transaction} />
+        ))
+      ) : (
+        <div className="no-transactions">
+          {t("No transactions found")}
```

Comment:
Let's define a new reference in  
https://github.com/vultisig/launch-web/blob/main/src/i18n/constant-keys.ts
and reuse it wherever needed. This will make it easier to update the text when necessary and also help us support multiple languages more efficiently.